samples/drivers/opamp/output_measure/src/main.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "main.h"
+#include <stdio.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/opamp.h>
+
+static int32_t sample_buffer;
+static const struct adc_dt_spec adc_channel =
+	ADC_DT_SPEC_GET(DT_PHANDLE(DT_PATH(zephyr_user), opamp));
+static const struct device *opamp_dev =
+	DEVICE_DT_GET(DT_PHANDLE(DT_PATH(zephyr_user), opamp));
+static const int32_t adc_read_delay_ms =
+	DT_PROP(DT_PATH(zephyr_user), adc_read_delay_ms);
+
+int main(void)
+{
+	int ret;
+
+	struct adc_sequence sequence = {
+		.buffer = &sample_buffer,
+		.buffer_size = sizeof(sample_buffer),
+	};
+
+	if (!adc_is_ready_dt(&adc_channel)) {
+		printf("ADC device %s is not ready\n", adc_channel.dev->name);
+		return 0;
+	}
+
+	if (!device_is_ready(opamp_dev)) {
+		printf("OPAMP device %s is not ready\n", opamp_dev->name);
+		return 0;
+	}
+
+	ret = adc_channel_setup_dt(&adc_channel);
+	if (ret < 0) {
+		printf("ADC channel could not setup with code (%d)\n", ret);
+		return 0;
+	}
+
+	ret = adc_sequence_init_dt(&adc_channel, &sequence);
+	if (ret < 0) {
+		printf("ADC sequence initialize failed with code (%d)\n", ret);
+		return 0;
+	}
+
+	ret = opamp_enable(opamp_dev);
+	if (ret < 0) {
+		printf("OPAMP enable failed with code (%d)\n", ret);
+		return 0;
+	}
+
+	if (adc_read_delay_ms > 0) {
+		k_msleep(adc_read_delay_ms);
+	}
+
+	ret = adc_read_dt(&adc_channel, &sequence);
+	if (ret < 0) {
+		printf("ADC read failed with code (%d)\n", ret);
+		return 0;
+	}
+
+	ret = adc_raw_to_millivolts(adc_channel.vref_mv,
+				adc_channel.channel_cfg.gain,
+				adc_channel.resolution, &sample_buffer);
+	if (ret < 0) {
+		printf("Convert to millivolts failed with code (%d)\n", ret);
+		return 0;
+	}
+
+	printf("OPAMP output is: %d(mv)\n", sample_buffer);
+
+#if OPAMP_SUPPORT_PROGRAMMABLE_GAIN
```

Comment:
To me the whole `#if ...` makes less sense here, since it is basically guards opamp_set_gain, which shall be callable in any case, since API functionality.
Better: Remove lines 69-77 and leave opamp_set_gain in place. Change for-loop to do-while one and set the gain after printing out the output. Use index counter for test_gain array indexing with different max values depending on the size of test_gain.

Reply:
I mean - function shall be always called, but decision if PGA is supported shall be made by driver itself - returning an error value if it is not implemented.